arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/MemoizedDeepRecursiveFunction.kt

```diff
@@ -2,6 +2,30 @@ package arrow.core
 
 import arrow.atomic.Atomic
 import arrow.atomic.loop
+import kotlin.jvm.JvmInline
+import kotlin.jvm.JvmOverloads
+
+public interface MemoizationCache<K, V> {
+  public fun get(key: K): V?
+  public fun set(key: K, value: V): V
```

Comment:
I think we can add `operator` modifier to `get` 🙏

```suggestion
  public operator fun get(key: K): V?
```

--> 
```
    when (val v = cache[x]) {
        ...
    }
```